src/nested_pandas/nestedframe/core.py

```diff
@@ -1111,6 +1111,85 @@ def explode(self, column: IndexLabel, ignore_index: bool = False):
+        self,
+        value: Hashable | Mapping | pd.Series | pd.DataFrame | None = None,
+        *,
+        axis: Axis | None = None,
+        inplace: bool = False,
+        limit: int | None = None,
+    ) -> NestedFrame | None:
+        """
+        Fill NA/NaN values using the specified method for base and nested columns.
+
+        Parameters
+        ----------
+        value : scalar, dict, Series, or DataFrame
+            Value to use to fill holes (e.g. 0), alternately a
+            dict/Series/DataFrame of values specifying which value to use for
+            each column.  Values not in the dict/Series/DataFrame will not be filled.
+            This value cannot be a list.
+        axis : {axes_single_arg}, default None
+            Axis along which to fill missing values.
+        inplace : bool, default False
+            If True, fill in-place. Note: this will modify any
+            other views on this object (e.g., a no-copy slice for a column in a
+            NestedFrame).
+        limit : int, default None
+            The maximum number of entries along the entire axis where NaNs will be
+            filled. Must be greater than 0 if not None. Currently, limit on nested
+            columns is not supported, meaning that all Nans will be filled (if there
+            is a value specified) regardless of the input.
+
+        Returns
+        -------
+        NestedFrame or None
+            NestedFrame with missing values filled or None if ``inplace=True``.
+
+        See Also
+        --------
+        :meth:`pandas.DataFrame.fillna`
+
+        Examples
+        --------
+        >>> import nested_pandas as npd
+        >>> nf = npd.NestedFrame(
+        ...     data={"a": [np.nan, 20, np.nan], "b": [np.nan, np.nan, 30], "c": [10, np.nan, np.nan]},
+        ...     index=[0, 1, 2]
+        ... )
+        >>> nested = pd.DataFrame(
+        ...     data={"d": [np.nan, np.nan, np.nan], "e": [np.nan, 1, np.nan]},
+        ...     index=[0, 1, 2]
+        ... )
+        >>> nf = nf.add_nested(nested, "nested")
+
+        >>> nf.fillna(0)
+              a     b     c              nested
+        0   0.0   0.0  10.0  [{d: 0.0, e: 0.0}]
+        1  20.0   0.0   0.0  [{d: 0.0, e: 1.0}]
+        2   0.0  30.0   0.0  [{d: 0.0, e: 0.0}]
+
+        """
+
+        if not self.nested_columns:
+            return super().fillna(value=value, axis=axis, inplace=inplace, limit=limit)
+
+        base_cols = [col for col in self.columns if col not in self.nested_columns]
+        filled_df = super().__getitem__(base_cols).fillna(value=value, axis=axis, inplace=False, limit=limit)
+
+        for nest_col in self.nested_columns:
+            nested_df = self[nest_col].nest.to_flat()
+            # prepend prefix to columns for value matching, undo after filled
+            nested_df.columns = [f"{nest_col}.{col}" for col in nested_df.columns]
+            nested_df = nested_df.fillna(value=value, axis=axis, inplace=False, limit=limit)
```

Comment:
If limit isn't supported for nested columns, let's set it to a static value (`None`) here to avoid unintended behavior that may occur from pandas or nested-pandas development in the future.